### PR TITLE
Refactor the shell file to typescript

### PR DESCRIPTION
**Pull request by wassertim**
This is the refactoring from js to typescript. Now we have types and everything stored in separate files.

@BrandonArp look how beautiful this is :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 *.class
 *.log
+.DS_Store
+src/main/resources/typescriptc.js
```
Comment (BrandonArp) [inline .gitignore]: Please remove this.  The file shouldn't exist after this change.

File: tsconfig.json [hunk 2]
```diff
@@ -7,9 +7,8 @@
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
     "removeComments": true,
-    "noImplicitAny": false,
+    "noImplicitAny": true,
     "noImplicitReturns": true,
-    "allowJs": true,
     "outFile": "src/main/resources/typescriptc.js"
```
Comment (BrandonArp) [inline tsconfig.json]: Is there a way we can build into the target directory and execute from there?  I'd prefer not to have code compiled into the source tree.

**Review by BrandonArp — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (wassertim) [reply .gitignore]: That's why I added it, so you will not accidentally commit it

File: tsconfig.json (hunk 2 above)
Comment (wassertim) [reply tsconfig.json]: If we move the resources folder outside of src it could do a trick, what do you think?

Comment (wassertim) [reply tsconfig.json]: I will try anyway

File: .gitignore (hunk 1 above)
Comment (wassertim) [reply .gitignore]: You are right, after I put the compilation result into the target - we don't need this anymore

File: tsconfig.json (hunk 2 above)
Comment (wassertim) [reply tsconfig.json]: This is done. Please review again

**Review by BrandonArp — APPROVED** [review]

**Comment by BrandonArp** [comment]
Excellent! Looks great!

**Pull request merged**